marshmallow_dataclass/__init__.py

```diff
@@ -117,6 +120,9 @@ def dataclass(
     order: bool = False,
     unsafe_hash: bool = False,
     frozen: bool = False,
+    match_args: bool = True,
+    kw_only: bool = False,
+    slots: bool = False,
     base_schema: Optional[Type[marshmallow.Schema]] = None,
     cls_frame: Optional[types.FrameType] = None,
 ) -> Callable[[Type[_U]], Type[_U]]:
```

Comment:
I feel these overloads should be conditional on python version.  I.e. The prototypes should not declare unsupported options.

Since (I think) we want `marshmallow_dataclass.dataclass` to have the same signature as `dataclasses.dataclass`, we should probably just copy typeshed's interface definition for `dataclass`.

https://github.com/python/typeshed/blob/8b9b0f4d031fe614a0211e1faadf57b3340e643e/stdlib/dataclasses.pyi#L57C1-L104C1.